17.0 fix merge content menu moc

d-fence opened the pull request:
1. [IMP] runbot: add a message when a build error is merged
    When digging into deactivated build errors, one cannot easily find why
    an error was deactivated and to which one it was merged.
2. [FIX] runbot: bind the link error content action to the right model
3. [IMP] runbot: add possibility to search errors for fingerprints
4. [IMP] runbot: find duplicate error contents
5. [IMP] runbot: reorganize build errors menus
6. [FIX] runbot: refactor error content merge
7. [IMP] runbot: rename _merge method into _relink on error_content
8. [IMP] runbot: add a deduplicate server action
9. [IMP] runbot: add responsible in error list view

d-fence committed:
[IMP] runbot: add a message when a build error is merged

When digging into deactivated build errors, one cannot easily find why
an error was deactivated and to which one it was merged.

With this commit, a message is added in the chatter to explain where it
is merged.

d-fence committed:
[FIX] runbot: bind the link error content action to the right model

d-fence committed:
[FIX] runbot: refactor error content merge

d-fence committed:
[IMP] runbot: add possibility to search fingerprints

d-fence committed:
[IMP] runbot: find duplicate error contents

Williambraecky commented on runbot/models/build_error.py:
```diff
@@ -511,6 +515,18 @@ def action_clean_content(self):
+    def _find_duplicates(self):
+        fingerprints = [r['fingerprint'] for r in self.env['runbot.build.error.content'].read_group([], ['id_count:count(id)', 'fingerprint'], ['fingerprint']) if r['id_count'] >1]
+        duplicates = self.env['runbot.build.error.content'].search([('fingerprint', 'in', fingerprints)], order='fingerprint')
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "runbot.build.error.content",
+            "domain": [('id', 'in', duplicates.ids)],
+            "context": {"create": False, 'group_by': ['fingerprint']},
+            "name": "Duplicate Error contents",
+            'view_mode': 'tree'
+        }
```
This achieves the same with a singular query to the database and less data transfer
```suggestion
        rg = self.env['runbot.build.error.content']._read_group(
            [],
            ['fingerprint'],
            ['id_count:count(id)', 'id_arr:array_agg(id)'],
            [('id_count', '>', 1)]
        )
        return {
            "type": "ir.actions.act_window",
            "res_model": "runbot.build.error.content",
            "domain": [('id', 'in', [content_id for r in rg for content_id in r[2]])],
            "context": {"create": False, 'group_by': ['fingerprint']},
            "name": "Duplicate Error contents",
            'view_mode': 'tree'
        }
```

Williambraecky reviewed (approved):
Minor optimisation :)

Xavier-Do commented on runbot/models/build_error.py:
```diff
@@ -487,10 +488,7 @@ def action_link_errors_contents(self):
-        self_sorted = self.sorted(lambda ec: (not ec.error_id.test_tags, not ec.error_id.responsible, ec.error_id.error_count, ec.id))
-        base_error = self_sorted[0].error_id
-        base_error._merge(self_sorted.error_id - base_error)
+        self._merge()
```
This is not correct, not the intention of the `action_link_errors_contents`, we want to merge parent, not errors.

d-fence committed:
[IMP] runbot: rename _merge method into _relink on error_content

d-fence committed:
[IMP] runbot: add a deduplicate server action

Duplicate error content should not happens ... but it does.
With this commit, a server actions allows to relink error contents and
thus removes error contents having the same fingerprint.

d-fence committed:
[IMP] runbot: add responsible in error list view

Xavier-Do committed:
[IMP] runbot: optimize build error content merging

d-fence committed:
[IMP] runbot: remove error logs menu
